From 626a8a703872cc46f74ff9bf7c7db0a306864f1b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tinja=20Paavosepp=C3=A4?= <tinja.paavoseppa@qt.io>
Date: Tue, 26 Nov 2024 14:20:52 +0200
Subject: [PATCH 19/20] Android: Quit hosting Context only when app is entirely
 Qt

When quitting Qt, only quit the hosting Activity or Service if it's
primarily a Qt app, i.e. Qt is in the driver's seat. When using QtQuick
for Android, i.e. Qt is just a View, we do not want to forcefully quit
Activities, that is up to the app code.

Task-number: QTBUG-123711
Pick-to: 6.8 6.9
Change-Id: Ib98700be8693c3d30a296998b863b0f224c9bf69
---
 src/plugins/platforms/android/androidjnimain.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/platforms/android/androidjnimain.cpp b/src/plugins/platforms/android/androidjnimain.cpp
index f90093a837c..b74e0fd359f 100644
--- a/src/plugins/platforms/android/androidjnimain.cpp
+++ b/src/plugins/platforms/android/androidjnimain.cpp
@@ -470,7 +470,7 @@ static void startQtApplication(JNIEnv */*env*/, jclass /*clazz*/)
             qWarning() << "dlclose failed:" << dlerror();
     }
 
-    if (m_applicationClass)
+    if (m_applicationClass && QtAndroid::isQtApplication())
         QJniObject::callStaticMethod<void>(m_applicationClass, "quitApp", "()V");
 
     sem_post(&m_terminateSemaphore);
-- 
2.48.1

